integration-tests/src/test/scala/play/libs/ws/ahc/AhcWSClientSpec.scala

```diff
@@ -99,10 +99,11 @@ class AhcWSClientSpec(implicit executionEnv: ExecutionEnv) extends Specification
         val responseXml = response.getBody(xml())
         responseXml.normalizeDocument()
 
-        responseXml.isEqualNode(document) must beTrue
+        responseXml.isEqualNode(document) must beTrue and {
```

Comment:
Sorry for not being clear.

I was not suggesting to use mocks to test this, but instead, have a real client (with a "real" server) testing the following scenarios:

1. A simple GET to a server responding HTTP status 200 should return the requested url
2. A simple GET to a server responding HTTP status 30x should return the final url when follow redirect is enabled
3. A simple GET to a server responding HTTP status 30x should return the requested url when follow redirect is disabled
4. Maybe a POST to a server responding HTTP status 307 should

WDYT?